src/osd/OSD.cc

```diff
@@ -9625,10 +9627,18 @@ void OSD::enqueue_op(spg_t pg, OpRequestRef&& op, epoch_t epoch)
   op->osd_trace.keyval("cost", cost);
   op->mark_queued_for_pg();
   logger->tinc(l_osd_op_before_queue_op_lat, latency);
-  op_shardedwq.queue(
-    OpSchedulerItem(
-      unique_ptr<OpSchedulerItem::OpQueueable>(new PGOpItem(pg, std::move(op))),
-      cost, priority, stamp, owner, epoch));
+  if (type == MSG_OSD_PG_PUSH ||
```

Comment:
should we include MSG_OSD_PG_PULL and MSG_OSD_PG_SCAN (for backfill) as well, or are these already handled elsewhere? Are other background work messages going through this path? It looks like there are a number of lower priority peering events (e.g. scrub requests) that would go in the immediate queue due to being queued as PGPeeringItems, though they aren't likely to be numerous enough to have the same effect as recovery ops.

Reply:
This is something that I am looking into. MSG_OSD_PG_SCAN does result in peering events getting enqueued and if they are of lower priority, it could be a candidate. I will test this like we discussed and determine if there are any significant effects.
I need to check MSG_OSD_PG_PULL as it appears to result in Push messages.